src/async_fuse/memfs/kv_engine/etcd_impl.rs

```diff
@@ -1,19 +1,209 @@
-use std::collections::HashMap;
 use std::fmt::Debug;
+use std::sync::atomic::AtomicBool;
 use std::time::Duration;
+use std::vec;
+use std::{collections::HashMap, sync::Arc};
 
 use async_trait::async_trait;
 use datenlord::metrics::KV_METRICS;
 use etcd_client::{
     Compare, CompareOp, DeleteOptions, GetOptions, LockOptions, PutOptions, Txn, TxnOp,
 };
+use tokio::sync::mpsc;
+use tracing::{error, info};
 
 use super::{
     check_ttl, conv_u64_sec_2_i64, fmt, DeleteOption, KVEngine, KeyType, KvVersion, LockKeyType,
     MetaTxn, SetOption, ValueType,
 };
 use crate::common::error::{Context, DatenLordResult};
 
+/// The keepalive session struct
+#[derive(Clone)]
+pub struct Session {
+    close_tx: mpsc::Sender<()>,
+    inner: Arc<SessionInner>,
+}
+
+impl Debug for Session {
+    #[inline]
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("Session").finish()
+    }
+}
+
+impl Session {
+    /// Create a new session
+    pub fn new(client: etcd_client::Client, ttl: i64, lease_id: i64) -> Arc<Self> {
+        let is_closed = Arc::new(AtomicBool::new(false));
+        let inner = Arc::new(SessionInner {
+            is_closed,
+            lease_id,
+            ttl,
+            client,
+        });
+        let (close_tx, close_rx) = mpsc::channel(1);
+
+        let _inner = inner.clone();
+        tokio::spawn(async move {
+            _inner.keep_alive(close_rx).await.unwrap();
+        });
+
+        let session = Session { close_tx, inner };
+
+        return Arc::new(session);
+    }
+
+    /// Get the lease id
+    pub fn lease_id(&self) -> i64 {
+        self.inner.lease_id
+    }
+
+    /// Get the ttl
+    pub fn ttl(&self) -> i64 {
+        self.inner.ttl
+    }
+
+    /// Get the is_closed flag
+    pub fn is_closed(&self) -> bool {
+        self.inner
+            .is_closed
+            .load(std::sync::atomic::Ordering::Relaxed)
```

Comment:
Should we really use `Relaxed`?

Reply:
No, you cannot use `AcqRel` at `load` or `store`. You should use `Acquire` at `load` and use `Release` at `store`.